pkg/sdkserver/sdkserver_test.go

```diff
@@ -985,6 +986,186 @@ func TestSDKServerPlayerCapacity(t *testing.T) {
 	agtesting.AssertEventContains(t, m.FakeRecorder.Events, "PlayerCapacity Set to 20")
 }
 
+func TestSDKServerPlayerConnectAndDisconnect(t *testing.T) {
+	t.Parallel()
+	agruntime.FeatureTestMutex.Lock()
+	defer agruntime.FeatureTestMutex.Unlock()
+
+	err := agruntime.ParseFeatures(string(agruntime.FeaturePlayerTracking) + "=true")
+	assert.NoError(t, err)
+
+	m := agtesting.NewMocks()
+	stop := make(chan struct{})
+	defer close(stop)
+
+	sc, err := defaultSidecar(m)
+	assert.Nil(t, err)
+
+	capacity := int64(3)
+	m.AgonesClient.AddReactor("list", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		gs := agonesv1.GameServer{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "test", Namespace: "default",
+			},
+			Spec: agonesv1.GameServerSpec{
+				SdkServer: agonesv1.SdkServer{
+					LogLevel: "Debug",
+				},
+				// this is here to give us a reference, so we know when sc.Run() has completed.
+				Players: &agonesv1.PlayersSpec{
+					InitialCapacity: capacity,
+				},
+			},
+		}
+		gs.ApplyDefaults()
+		return true, &agonesv1.GameServerList{Items: []agonesv1.GameServer{gs}}, nil
+	})
+	updated := make(chan *agonesv1.PlayerStatus, 10)
+	m.AgonesClient.AddReactor("update", "gameservers", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		ua := action.(k8stesting.UpdateAction)
+		gs := ua.GetObject().(*agonesv1.GameServer)
+		updated <- gs.Status.Players
+		return true, gs, nil
+	})
+
+	sc.informerFactory.Start(stop)
+	assert.True(t, cache.WaitForCacheSync(stop, sc.gameServerSynced))
+
+	go func() {
+		err = sc.Run(stop)
+		assert.NoError(t, err)
+	}()
+
+	// check initial value comes through
+	// async, so check after a period
+	e := &alpha.Empty{}
+	err = wait.Poll(time.Second, 10*time.Second, func() (bool, error) {
+		count, err := sc.GetPlayerCapacity(context.Background(), e)
+		return count.Count == capacity, err
+	})
+	assert.NoError(t, err)
+
+	count, err := sc.GetPlayerCount(context.Background(), e)
+	assert.Equal(t, int64(0), count.Count)
+
+	list, err := sc.GetConnectedPlayers(context.Background(), e)
+	assert.NoError(t, err)
+	assert.Empty(t, list.List)
+
+	ok, err := sc.IsPlayerConnected(context.Background(), &alpha.PlayerID{PlayerID: "1"})
+	assert.NoError(t, err)
+	assert.False(t, ok.Bool, "no player connected yet")
+
+	// sdk value should always be correct, even if we send more than one update per second.
+	for i := 0; i < 3; i++ {
```

Comment:
Probably it would be easier to understand why `3` (if I got it right):
```suggestion
	for i := 0; i < capacity; i++ {
```